pkg/ingester/ingester.go

```diff
@@ -1690,8 +1697,10 @@ func (i *Ingester) getOrCreateTSDB(userID string, force bool) (*userTSDB, error)
 	return db, nil
 }
 
+type tsdbOption func(*tsdb.Options)
+
 // createTSDB creates a TSDB for a given userID, and returns the created db.
-func (i *Ingester) createTSDB(userID string) (*userTSDB, error) {
+func (i *Ingester) createTSDB(userID string, additionalTsdbOptions ...tsdbOption) (*userTSDB, error) {
```

Comment:
Passing `tsdbOption` looks overkilled. You can directly pass `int walReplayConcurrency`. When called from `getOrCreateTSDB()` you can just pass `0`.